lib/charms/prometheus_k8s/v1/prometheus.py

```diff
@@ -317,99 +314,71 @@ def __init__(self, charm, name, consumes, multi=False):
                 dictionary are corresponding minimal acceptable
                 semantic version specfications for the monitoring
                 service.
+            service: string name of Pebble service of consumer charm.
+            config: a optional dictionary with keys that are
+                Prometheus scrape configuration options, for example
+                metrics path and port.
             multi: an optional (default False) flag to indicate if
                 this object must support interaction with multiple
                 Prometheus monitoring service providers.
-
         """
         super().__init__(charm, name, consumes, multi)
+
         self._charm = charm
+        self._service = service
         self._relation_name = name
-        self._stored.set_default(targets={})
+        self._static_scrape_port = config.get("static_scrape_port", 80)
+        self._static_scrape_path = config.get("static_scrape_path", "/metrics")
```

Comment:
Thanks for pointing this out. Give me some time to digest this. Kube state metrics operator is an important one so I will study it and reach out to you, so we can explore what our options are and what works best. If required I will consider further refactors in subsequent commits.

Reply:
From discussion with @johnsca it turns out the Kube State Metrics use case is that each unit of that charm exposes two endpoints (paths+ports). The units are homogenous. The next PR will support multiple static scrape config metadata blocks. The unit IPs will still be gathered directly from each unit. Each of these gathered IPs will be used to populate each of the multiple [static scrape config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#static_config) metadata blocks. This will support @johnsca use case and many more.

Reply:
My previous post was not clear. Let me try and clarify. Prometheus [scrape configuration](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config) allows for a list of `<scrape_config>` blocks, each representing one "job". Each job has multiple metadata fields including `metrics_path`.  This implies each job can only have the same metrics path. Since @johnsca use case does conform to a single metrics path, the next PR will support only this, but it may be extended to multiple metrics paths using multiple jobs in the future. This forthcoming PR will allow units set one or more "metric ports" by a mechanism as yet not finalized.

Reply:
I have created a [ticket](https://github.com/canonical/prometheus-operator/issues/73) to pick up this feature in the next PR. Resolving this conversation here form. We can continue requirements gathering on the cited ticket.